### ISSUE TITLE
pipeline view display stages twice

### ISSUE THREAD

**Issue by sancelot** [jenkinsci/gitlab-plugin#771]
### Context
- **Gitlab plugin version**: 1.5.6
- **Gitlab version**: 10.5.3
- **Jenkins version**: 2.107.3
- **Job type**: Pipeline

### Problem description
When a jenkins pileine job has been or is running , the pipeline stages status are displayed twice.
![screenshot](https://user-images.githubusercontent.com/122811/40790407-24f03abc-64f5-11e8-87ad-cbbdf7d94b99.JPG)

[Jenkinsfile](https://github.com/jenkinsci/gitlab-plugin/files/2058772/Jenkinsfile.txt)

furthermore, May be there is a way to display the full text .

**Comment by omehegan** [comment]
@sancelot hmm, this is interesting. I notice in your screenshot that 'Checkout sources' is also shown, yet you have not wrapped that in a `gitlabBuilds` or `gitlabCommitStatus` block. I wonder if every stage's status is being sent automatically... Not sure why that would be happening. Can you follow the instructions in the User Support section of the README to increase debug logging for the plugin, and then attach those logs after you reproduce this issue? We can see if the plugin is sending two discrete events, or possibly one event with a map and then a bunch of discrete events. Also looking at the Pipeline Steps log in Jenkins might show more clearly where this is happening.

**Comment by omehegan** [comment]
@sancelot also I wonder if you just don't need to use both gitlabBuilds and gitlabCommitStatus, even though that is what the doc says to do. You could try removing the gitlabCommitStatus parts and see...

**Comment by sancelot** [comment]
I   have not wrapped 'check out sources' because I read it in the manual... 

Removing gitlabcommitstatus has not helped. In this case I noticed that at first all the stages  were displayed and then the running stages appended To the list 
I Will try providing a log in the day

**Comment by sancelot** [comment]
[gitlab_jenkinslog.txt](https://github.com/jenkinsci/gitlab-plugin/files/2148349/gitlab_jenkinslog.txt)
log file added

**Comment by semos** [comment]
@omehegan after a first log review, all statuses prepended by a #number seems to be sent automatically.

The first is the "checkout sources", which is not wrapped nor configured to be sent in gitlabBuilds.

Then, during the build, normal statuses are sent with the gitlabCommitStatus.

Once the build is complete, all other statuses "#xxx" are sent in a row.

@sancelot i would advise to use multiple `updateGitlabCommitStatus name: 'build', state: 'pending'` at the start of your jenkins file if you want all statuses to be sent as pending on the pipeline launch

**Comment by omehegan** [comment]
Yeah it's definitely the plugin sending things twice. I think I see what is happening. It is sending:

```
pending: "Compiling linux64"
running: "#2 Compiling linux64"
success: "#2 Compiling linux64"
```

So it seems like the pending status string is different from the running and success statuses, so it shows up twice in GitLab. 

@ttsalonen do you think this is related to #693?

**Comment by ttsalonen** [comment]
@omehegan I don't think this is related to https://github.com/jenkinsci/gitlab-plugin/pull/693.

I wonder where the "#2 Compiling linux64" and "Checking out sources" statuses come from? Is it possible that there is also some other Jenkins job that sends those statuses to Gitlab? Are there multiple webhooks configured in Gitlab?

I tested the attached Jenkinsfile and one "duplicate" build status is caused by different build names used for "Compiling linux 32bits" stage. In gitlabBuilds section it is "Compiling linux 32bits" and later in gitlabCommitStatus step  it is "Compiling linux 32 bits" (space between 32 and bits). Other than that the Jenkinsfile seems to be working for me as expected.

**Comment by sancelot** [comment]
I corrected the "32bits" space mistake.

The problem disappear if I change the Jenkins file as follow :  
```
  gitlabBuilds(builds: [
                            "#2 Compiling linux64", 
                            "#3 Compiling linux 32 bits",
                            "#4 Compiling win32",
                            "#5 Packaging des sources python pour win32",
                            "#6 Static code analysis",
                            "#7 Tests unitaires",
                            "#8 Scan for warnings",
                            "#9 Deploy"
                            ]) {
```

**Comment by omehegan** [comment]
@ttsalonen thanks for looking at this!

@sancelot I don't see why you would need to number the stages. I think @ttsalonen has a good theory, is there another job also being triggered by this git repo? If you look at the plugin debug log now, I have a feeling you will see multiple updates which now just have identical strings, and so GitLab is creating one entry and then updating it with another push comes through. Either that, or this is some kind of GitLab behavior where it numbers the updates, and you're just matching that. What do you think?

**Comment by sancelot** [comment]
You are right, that is a silly hack and adding #number does not solve the problem.

I don't really know where to look on my side, you need some jenkins setup files ? Please can you tell me where I must look at.
Some branches are pulled and compiled on the same repo . The branch compiled is not the master.
jenkins polls the repository every 2 hours (this means it may wait for end of some jobs).
Only one jenkins job is being  run , other jobs will wait on the queue, till it is ready.

The pipeline is a pipeline "from multibranch configuration"
jenkins setup:
![jenkins_gitlab_setup](https://user-images.githubusercontent.com/122811/42384123-73bb65c6-8139-11e8-816d-448517690e61.jpg)

**Comment by omehegan** [comment]
@sancelot I don't recognize that UI in your last screenshot. Is it possible you have another GitLab plugin installed, like GitLab Branch Source or something?

**Comment by sancelot** [comment]
Yes I have gitlab  branch source plugin too

**Comment by omehegan** [comment]
@sancelot I don't think that plugin is supported or really working yet. I would recommend that you uninstall that if possible.

@Argelbargel was working on that, but the intent was to merge its code into this plugin. That has been on hold for awhile though, due to him being too busy to work on it.

**Comment by sancelot** [comment]
Hi guys, 
If setted up gitlab branch source plugin, in order to pull the whole gitlab changes.
in this case, if I remember, I would have to setup manually a project in jenkins for each gitlab branch isn't it ?

**Comment by omehegan** [comment]
@sancelot you can create Pipeline Multibranch projects, which will automatically add/remove sub-projects for each *branch* of a single repo. That much is supported and documented in our README. The GitLab Branch Source plugin tries to support creating new jobs for each new *repo* that is created in GitLab. But AFAIK that plugin is incomplete and not supported, so I would not recommend trying to use it.

**Comment by sancelot** [comment]
I removed the gitlab branch source plugin and created a pipeline multibranch project
Unfortunately the issue is always present :-(

**Comment by omehegan** [comment]
@sancelot can you capture logs again for the run of this new job you created? I'm still trying to understand why the plugin seems to be sending 2x updates for your jobs, when this is not an issue for anyone else.

**Comment by sancelot** [comment]
We finally rebooted the gitlab server. I don't know if there was some cached data, but the problem is no more present. 

The only thing noticed is that the displayed labels order does not correspond to the execution order.

**Comment by omehegan** [comment]
@sancelot OK, I will close this issue. Please reopen if you find the issue occurring again.
